openrtx/include/core/state.h

```diff
@@ -62,6 +62,17 @@ typedef struct
     bool       restore_eflash;
     bool       txDisable;
     uint8_t    step_index;
+    #ifdef CONFIG_SPECTRUM
+    bool       spectrum_shouldRefresh; // True if one scanning cycle is done.
+    uint32_t   spectrum_startFreq; // Start frequency for spectrum display, in 10Hz units
+    uint32_t   spectrum_peakFreq;  // Current peak frequency for spectrum display, in 10Hz units
```

Comment:
I'd move all the return values from the rtx, that is peak values, samples, ... to a dedicated data structure like `struct rxSweep` defined inside rtx.h An instance of this struct struct can then be part of the `rtxStatus_t` struct.

Also, don't use magic numbers for the size of the array, use a macro